ts/DataGrid/Table/Actions/RowsVirtualizer.ts

```diff
@@ -244,17 +256,30 @@ class RowsVirtualizer {
      */
     private renderRows(rowCursor: number): void {
         const { viewport: vp, buffer } = this;
-        const rowsPerPage = Math.ceil(
+        const isVirtualization = this.rowSettings?.virtualization;
+        const rowsPerPage = isVirtualization ? Math.ceil(
             vp.tbodyElement.offsetHeight / this.defaultRowHeight
-        );
+        ) : Infinity; // need to be refactored when add pagination
 
         const rows = vp.rows;
 
+        if (!isVirtualization && rows.length > 50) {
+            console.warn(
+                'DataGrid: a large dataset can cause performance ' +
+                'issues in sorting or resizing columns'
```

Comment:
Do we need to specify that only `in sorting or resizing`? IMO we are unnecessarily narrowing the scope in the warning message, because performance issues can also appear during initial rendering, editing, all other cases of querying, etc.

Reply:
Good point, let's replace it with the general warn.